frontend/web/components/CodeHelp.js

```diff
@@ -102,13 +102,14 @@ const CodeHelp = class extends Component {
 
   render() {
     const { hideHeader } = this.props
-    const language =
+    let language =
       this.state.language ||
       flagsmith.getTrait('preferred_language') ||
       Object.keys(this.props.snippets)[0]
     const tab = language
       ? Math.max(Object.keys(this.props.snippets).indexOf(language), 0)
       : 0
+    language = Object.keys(this.props.snippets)[tab]
```

Comment:
This area should be commented since the logic and underlying reasoning is non-obvious.